kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterServiceDependentResource.java

```diff
@@ -72,28 +73,49 @@ public Map<String, Service> desiredResources(
                 .flatMap(ProxyNetworkingModel.ClusterNetworkingModel::requiredSniLoadbalancerPorts)
                 .distinct().sorted().toList();
 
-        var sniServiceStream = sniLoadbalancerServices(primary, sharedSniLoadbalancerPorts);
+        Set<BootstrapServersAnnotation.BootstrapServer> bootstraps = getLoadBalancerServiceBootstrapServers(clusterNetworkingModels);
+
+        var sniServiceStream = sniLoadbalancerServices(primary, sharedSniLoadbalancerPorts, bootstraps);
 
         return Stream.concat(serviceStream, sniServiceStream).collect(toByNameMap());
     }
 
-    private ObjectMeta serviceMetadata(KafkaProxy primary, String name) {
-        return new ObjectMetaBuilder()
+    /**
+     * Get the bootstrap servers hosted by the shared LoadBalancer Service
+     */
+    private static Set<BootstrapServersAnnotation.BootstrapServer> getLoadBalancerServiceBootstrapServers(List<ProxyNetworkingModel.ClusterNetworkingModel> clusterNetworkingModels) {
+        return clusterNetworkingModels.stream()
+                .flatMap(ClusterServiceDependentResource::getBootstrapServers)
+                .collect(Collectors.toSet());
+    }
+
+    private static Stream<BootstrapServersAnnotation.BootstrapServer> getBootstrapServers(ProxyNetworkingModel.ClusterNetworkingModel networking) {
+        return networking.clusterIngressNetworkingModelResults().stream()
+                .map(ProxyNetworkingModel.ClusterIngressNetworkingModelResult::clusterIngressNetworkingModel)
+                .filter(clusterIngressModel -> clusterIngressModel.requiredSniLoadBalancerServicePorts()
+                        .findAny().isPresent())
+                .map(clusterIngressModel -> new BootstrapServersAnnotation.BootstrapServer(ResourcesUtil.name(clusterIngressModel.cluster()),
+                        ResourcesUtil.name(clusterIngressModel.ingress()), clusterIngressModel.bootstrapServers()));
```

Comment:
It feels a bit strange to filter based on `requiredSniLoadBalancerServicePorts` and then not mention ports here. are they implicitly part of `bootstrapServers` ?

Reply:
all ClusterIngressNetworkModels have a bootstrapServers, but we only want to add them to the LoadBalancerService if the model is exposed via that shared service. 

Maybe it means we should be returning an object like `SharedLoadBalancerRequirements` that declares "i need these client-facing ports, and these are the bootstrap servers I expect clients will connect with".

Reply:
Have introduced:

```
Optional<SharedLoadBalancerServiceRequirements> ClusterIngressNetworkingModel#sharedLoadBalancerServiceRequirements()
```

which, if present, defines the requirements of the shared service. Which client facing ports are required, and a `ClusterIngressBootstrapServer` to be added to the annotations.

the clusterIP implementations no longer need to implement `String bootstrapServers()`